src/InMemoryDocumentStore.php

```diff
@@ -516,4 +537,39 @@ private function sort(&$docs, OrderBy $orderBy)
             return $docCmp($docA, $docB, $orderBy);
         });
     }
+
+    /**
+     * The internal array_replace_recursive function also replaces sequential arrays recursively. This method aims to
+     * behave identical to array_replace_recursive but only when dealing with associative arrays. Sequential arrays
+     * are handled as if they were scalar types instead.
+     * 
+     * @param array $array1
+     * @param array $array2
+     * @return array
+     */
+    private function arrayReplaceRecursiveAssocOnly(array $array1, array $array2): array
```

Comment:
@jsor @heiglandreas can you take a look at these changes as well? It only affects testing environment and should not cause any existing tests to fail, but at least you should know about the change ;)